### DISCUSSION TITLE
Axiom scan triggered/run from Python

### DISCUSSION THREAD

**Discussion by proveyoureahuman** [pry0cc/axiom#790] [Q&A] [answered]
Hi, 

I need some help running an axiom scan from python. I have tried so many different ideas, but I just can't get it to work. 

If I run the bash script from the command line, everything is great. 

However, everything I have tried from python has failed. Does anyone have any ideas how to get in to work? 

Many thanks for any ideas.

./bash_script.sh
```
#!/bin/bash
/home/linux/.axiom/interact/axiom-scan  /home/linux/Desktop/input.txt -m waybackurls -o /home/linux/Desktop/output.txt
```

./python_script.py
```
#!/usr/bin/env python3
import subprocess
cmd = "/home/linux/Desktop/bash_script.sh"
p = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
#stdout, stderr = p.communicate()
for line in p.stdout:
    print (line.decode('utf8'))
p.wait()
print (p.returncode)
```

> // Output
> ==> /home/linux/.axiom/tmp/waybackurls+171072099419856/logs/stream01 <==
> bash: command: No such file or directory
> ==> /home/linux/.axiom/tmp/waybackurls+171072099419856/logs/stream02 <==
> bash: command: No such file or directory

**Comment by pdelteil** [comment]
Hello there,

Why would you need to run a bash script from python?

**Comment by proveyoureahuman** [comment]
Thanks - I need to trigger the process from Python so that it fits in with my current automation that is Python based. Many thanks.

**Comment by proveyoureahuman** [comment, marked as answer]
figured it out :)
